### DISCUSSION TITLE
Debugging missing feature in assemble repository step

### DISCUSSION THREAD

**Discussion by kjkeefe** [eclipse-tycho/tycho#1123] [Q&A]
I followed the example itp04 to build pom files for all my plugins, features, and products. https://github.com/eclipse/tycho/tree/master/demo/itp04-rcp

Everything appears to go well until the default-assemble-repository step of the product. I am getting this error message:

[INFO] --- tycho-p2-repository-plugin:2.7.4:assemble-repository (default-assemble-repository) @ product.core ---
[INFO] {osgi.os=win32, osgi.ws=win32, org.eclipse.update.install.features=true, osgi.arch=aarch64}
[ERROR] Cannot resolve project dependencies:
[ERROR]   Software being installed: edu.illinois.mobius.product.core 2.6.0.qualifier
[ERROR]   Missing requirement: edu.illinois.mobius.product.core 2.6.0.qualifier requires 'org.eclipse.equinox.p2.iu; edu.illinois.mobius.feature.feature.group 0.0.0' but it could not be found
[ERROR]
[ERROR] See https://wiki.eclipse.org/Tycho/Dependency_Resolution_Troubleshooting for help.

I read the wiki page and understand that p2 is not finding the edu.illinois.mobius.feature feature. However, when I go and look at the result of that step, it looks like it is being built. Here is the output from that step:

[INFO] ----------< edu.illinois.mobius:edu.illinois.mobius.feature >-----------
[INFO] Building edu.illinois.mobius.feature 2.6.0-SNAPSHOT              [49/79]
[INFO] --------------------------[ eclipse-feature ]---------------------------
[INFO]
[INFO] --- tycho-packaging-plugin:2.7.4:build-qualifier-aggregator (default-build-qualifier-aggregator) @ edu.illinois.mobius.feature ---
[INFO] The project's OSGi version is 2.6.0.202207110553
[INFO]
[INFO] --- tycho-packaging-plugin:2.7.4:validate-id (default-validate-id) @ edu.illinois.mobius.feature ---
[INFO]
[INFO] --- tycho-packaging-plugin:2.7.4:validate-version (default-validate-version) @ edu.illinois.mobius.feature ---
[INFO]
[INFO] --- target-platform-configuration:2.7.4:target-platform (default-target-platform) @ edu.illinois.mobius.feature ---
[INFO]
[INFO] --- tycho-packaging-plugin:2.7.4:update-consumer-pom (default-update-consumer-pom) @ edu.illinois.mobius.feature ---
[INFO]
[INFO] --- tycho-packaging-plugin:2.7.4:package-feature (default-package-feature) @ edu.illinois.mobius.feature ---
[INFO] Building jar: /Users/kjkeefe/work/gitlab/mobius/features/edu.illinois.mobius.feature/target/edu.illinois.mobius.feature-2.6.0-SNAPSHOT.jar
[INFO]
[INFO] --- tycho-p2-plugin:2.7.4:p2-metadata-default (default-p2-metadata-default) @ edu.illinois.mobius.feature ---
[INFO]
[INFO] --- tycho-p2-plugin:2.7.4:feature-p2-metadata (default-feature-p2-metadata) @ edu.illinois.mobius.feature —

How can I debug this further? Where should the produced feature be on the disk other than the jar file in the output? 

Thanks in advance!

**Comment by laeubi** [comment]
Could you share a runnable example of this?

The only unusual in the output I can see is:
`[INFO] {osgi.os=win32, osgi.ws=win32, org.eclipse.update.install.features=true, osgi.arch=aarch64}`

Is `aarch64` really a supported architecture for `win32`?

**Reply by kjkeefe** [reply]
I haven't been able to recreate it in code that I can release. I was hoping there was some way to examine where p2 caches the package during that step and see what is missing or in the wrong place.
